src/app/games/diaballik/tests/diaballik.component.spec.ts

```diff
@@ -0,0 +1,414 @@
+import { fakeAsync } from '@angular/core/testing';
+import { ComponentTestUtils } from 'src/app/utils/tests/TestUtils.spec';
+import { DiaballikComponent } from '../diaballik.component';
+import { DiaballikMove, DiaballikBallPass, DiaballikTranslation } from '../DiaballikMove';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { Coord } from 'src/app/jscaip/Coord';
+import { DiaballikPiece, DiaballikState } from '../DiaballikState';
+import { RulesFailure } from 'src/app/jscaip/RulesFailure';
+import { DiaballikFailure } from '../DiaballikFailure';
+
+describe('DiaballikComponent', () => {
+
+    let testUtils: ComponentTestUtils<DiaballikComponent>;
+
+    const O: DiaballikPiece = DiaballikPiece.ZERO;
+    const Ȯ: DiaballikPiece = DiaballikPiece.ZERO_WITH_BALL;
+    const X: DiaballikPiece = DiaballikPiece.ONE;
+    const Ẋ: DiaballikPiece = DiaballikPiece.ONE_WITH_BALL;
+    const _: DiaballikPiece = DiaballikPiece.NONE;
+
+    beforeEach(fakeAsync(async() => {
+        testUtils = await ComponentTestUtils.forGame<DiaballikComponent>('Diaballik');
+    }));
+    it('should create', () => {
+        testUtils.expectToBeCreated();
+    });
+    it('should finish the move when clicking on the done button after one sub move', fakeAsync(async() => {
+        // Given a state with a submove already done
+        await testUtils.expectClickSuccess('#click_0_6');
+        await testUtils.expectClickSuccess('#click_0_5');
+
+        // When clicking on the 'done' button
+
+        // Then it should succeed
```

Comment:
Petiote convention, je propose que le when contienne la description du move.
Et si y a rien à mettre dans le when, alors pas the when \n\n then mais  when \n then (collé quoi)

Reply:
Ehhh j'ai déjà aéré le tout :D Hésite pas à commenter là où y'a du newline en trop du coup !